src/Microsoft.AspNetCore.Razor/Parser/TagHelpers/TagHelperParseTreeRewriter.cs

```diff
@@ -35,10 +39,12 @@ public class TagHelperParseTreeRewriter : ISyntaxTreeRewriter
             "wbr"
         };
 
-        private TagHelperDescriptorProvider _provider;
-        private Stack<TagBlockTracker> _trackerStack;
+        private readonly List<KeyValuePair<string, string>> _htmlAttributeTracker;
+        private readonly StringBuilder _attributeValueBuilder;
+        private readonly TagHelperDescriptorProvider _provider;
+        private readonly Stack<TagBlockTracker> _trackerStack;
+        private readonly Stack<BlockBuilder> _blockStack;
```

Comment:
The lifetime of these is the parse of a single Razor page. If you feel strongly about this please file an issue.